src/components/AppLayout/Header/components/Layout.tsx

```diff
@@ -90,6 +121,42 @@ const Layout = ({ classes, providerDetails, providerInfo }) => {
           </Popper>
         )}
       />
+      <NetworkSelector
+        open={openNetworks}
+        selected={networkName}
+        toggle={toggleNetworks}
+        render={(networkRef) => (
+          <Popper
+            anchorEl={networkRef.current}
+            className={classes.popper}
+            open={openNetworks}
+            placement="bottom"
+            popperOptions={{ positionFixed: true }}
+          >
+            {({ TransitionProps }) => (
+              <Grow {...TransitionProps}>
+                <>
+                  <ClickAwayListener mouseEvent="onClick" onClickAway={clickAwayNetworks} touchEvent={false}>
+                    <List className={classes.network} component="div">
+                      {networks.map((network) => (
+                        <React.Fragment key={network.id}>
+                          <StyledLink href={network.safeUrl}>
+                            <NetworkLabel networkInfo={network} networkName={network.label} />
+                            {sameString(networkName, network.label.toLowerCase()) && (
+                              <Icon type="check" size="md" color="primary" />
+                            )}
+                          </StyledLink>
+                          <StyledDivider />
+                        </React.Fragment>
+                      ))}
+                    </List>
+                  </ClickAwayListener>
+                </>
+              </Grow>
+            )}
+          </Popper>
+        )}
+      />
```

Comment:
I would expect the NetworkSelector component itself to contain all of that code.
Can we move it there?

Reply:
Agree, but it looks like it's following the logic from the previous component `Provider`, maybe extracting both?